mathics/builtin/patterns.py

```diff
@@ -621,9 +670,14 @@ def yield_match(vars_2, rest):
             else:
                 yield_func(vars_2, None)
 
-        self.pattern.match(yield_match, expression, vars, evaluation)
+        pattern_context = pattern_context.copy()
+        pattern_context["yield_func"] = yield_match
+        self.pattern.match(expression, pattern_context)
+
+    def match_real_numberq(self, expression: Expression, pattern_context: dict):
+        """Match function for RealValudNumberQ"""
```

Comment:
RealValudNumberQ -> RealValuedNumberQ   and will be in next commit of more-patterns-more